File: elixir/hello-world/hello_world.exs
defmodule HelloWorld do
  @doc """
  Greets the user by name, or by saying "Hello, World!"
  if no name is given.
  """
  @spec hello(String.t) :: String.t
  def hello(name \\ "World") do
    "Hello, #{name}!"
  end
end

File: kotlin/hello-world/src/main/kotlin/HelloWorld.kt
object HelloWorld {
  fun hello(maybeName: String? = "World"): String {
    val name: String

    if (maybeName.isNullOrBlank()) {
      name = "World"
    } else {
      name = maybeName!!
    }

    return "Hello, $name!"
  }
}
